src/masonite/cache/drivers/RedisDriver.py

```diff
@@ -0,0 +1,99 @@
+import json
+
+
+class RedisDriver:
+    def __init__(self, application):
+        self.application = application
+        self.connection = None
+
+    def set_options(self, options):
+        self.options = options
+        return self
+
+    def get_connection(self):
+        try:
+            import redis
+        except ImportError:
+            raise ModuleNotFoundError(
+                "Could not find the 'redis' library. Run 'pip install redis' to fix this."
+            )
+
+        if not self.connection:
+            self.connection = redis.StrictRedis(
+                host=self.options.get("host"),
+                port=self.options.get("port"),
+                password=self.options.get("password"),
+                decode_responses=True,
+            )
+
+        return self.connection
+
+    def add(self, key, value):
+        if self.has(key):
+            return self.get(key)
+
+        self.put(key, value)
+        return value
+
+    def get(self, key, default=None, **options):
+        if not self.has(key):
+            return default
+        return self.get_value(
+            self.get_connection().get(f"{self.get_name()}_cache_{key}")
+        )
+
+    def put(self, key, value, seconds=None, **options):
+
+        time = self.get_expiration_time(seconds)
+
+        if isinstance(value, (dict, list)):
+            value = json.dumps(value)
+
+        return self.get_connection().set(
+            f"{self.get_name()}_cache_{key}", value, ex=time
+        )
+
+    def has(self, key):
+        return self.get_connection().get(f"{self.get_name()}_cache_{key}")
+
+    def increment(self, key, amount=1):
+        return self.put(key, str(int(self.get(key)) + amount))
+
+    def decrement(self, key, amount=1):
+        return self.put(key, str(int(self.get(key)) - amount))
+
+    def remember(self, key, callable):
+        value = self.get(key)
+
+        if value:
+            return value
+
+        callable(self)
+
+    def forget(self, key):
+        try:
+            self.get_connection().delete(f"{self.get_name()}_cache_{key}")
+            return True
+        except FileNotFoundError:
```

Comment:
same remark here, I guess it's okay for FileDriver but feels weird for RedisDriver ?